### DISCUSSION TITLE
TensorLy prediction NaN

### DISCUSSION THREAD

**Discussion by QhL-UNC** [tensorly/tensorly#337] [Q&A]
I recently need to use tensor regression for my project. The predictors are in the following format: Number of Obs * Number of Features * Number of Days, and the response is a scalar: a numeric response. 
I changed my data into number of obs * number of features * number of days using tl.tensor() function, and fit a CPRegressor(), and all the predicted values are NaNs. 

Anyone has any ideas?

**Comment by JeanKossaifi** [comment]
Can you post a minimal snippet of code to reproduce your issue?
Have you tried centering your data?

**Reply by QhL-UNC** [reply]
# Scale the Word Cloud Features
from sklearn.preprocessing import StandardScaler
scaler = StandardScaler().fit(demo_data)
demo_data_scaled = scaler.transform(demo_data)

scaler2 = StandardScaler().fit(wc_data)
wc_data_scaled = scaler2.transform(wc_data) 
wc_data_scaled = wc_data_scaled.reshape(1003, 97, 14)


# Change data into tensor format
X = tl.tensor(wc_data_scaled)
X.shape  
### X shape: 1003 (Number of obs), 97 (Number of Features), 14 (Number of Days for each feature)

# Fit model
from tensorly.regression.tucker_regression import TuckerRegressor
TG_model = TuckerRegressor(weight_ranks=[2, 3], reg_W=2, n_iter_max=200, verbose=2)

TG_fit = TG_model.fit(X, Y)
### Y is a numerical depression score

# Prediction
TG_fit.predict(X)
### the results are all "nan"
